crates/oxc_linter/src/rules/eslint/eq_eq_eq.rs

```diff
@@ -1,21 +1,61 @@
-use oxc_ast::{ast::Expression, AstKind};
+use oxc_ast::{
+    ast::{BinaryExpression, Expression},
+    AstKind,
+};
 use oxc_diagnostics::{
     miette::{self, Diagnostic},
     thiserror::{self, Error},
 };
 use oxc_macros::declare_oxc_lint;
-use oxc_span::Span;
+use oxc_span::{GetSpan, Span};
 use oxc_syntax::operator::{BinaryOperator, UnaryOperator};
 
 use crate::{context::LintContext, fixer::Fix, rule::Rule, AstNode};
 
 #[derive(Debug, Error, Diagnostic)]
 #[error("eslint(eqeqeq): Expected {1} and instead saw {0}")]
-#[diagnostic(severity(warning), help("Prefer strict {1} operator"))]
+#[diagnostic(severity(warning), help("Prefer {1} operator"))]
 struct EqEqEqDiagnostic(&'static str, &'static str, #[label] pub Span);
 
 #[derive(Debug, Default, Clone)]
-pub struct EqEqEq;
+enum CompareType {
+    #[default]
+    Always,
+    Smart,
+}
+
+impl CompareType {
+    pub fn from(raw: &str) -> Self {
+        match raw {
+            "smart" => Self::Smart,
+            _ => Self::Always,
+        }
+    }
+}
+
+#[derive(Debug, Default, Clone)]
+enum NullType {
+    #[default]
+    Always,
+    Never,
+    Ignore,
+}
+
+impl NullType {
+    pub fn from(raw: &str) -> Self {
+        match raw {
+            "always" => Self::Always,
+            "never" => Self::Never,
+            _ => Self::Ignore,
+        }
+    }
+}
```

Comment:
nit: move these code after `impl Rule for EqEqEq` to make this file easier to ready